docs/diagrams/Recommender.puml

```diff
@@ -0,0 +1,18 @@
+@startuml
+'https://plantuml.com/class-diagram
+!include style.puml
+skinparam arrowThickness 1.1
+skinparam arrowColor LOGIC_COLOR
+skinparam classBackgroundColor LOGIC_COLOR
+
+
+package Recommender {
+    class Recommender
+    class TimingRecommender
+    class LocationRecommender
+}
+
+LocationRecommender -u-> Recommender
+TimingRecommender -u-> Recommender
+LocationRecommender ..> TimingRecommender
```

Comment:
Ok I’ve made it such that the two recommenders don’t rely on each other. TimingRecommender returns a list of time periods so we don’t really need it anymore. LocationRecommender is initialised with the persons, so there’s where we get the timetables.

Reply:
Why am I deleting TimingRecommender again?

Reply:
> TimingRecommender returns a list of time periods so we don’t really need it anymore.

Why am I deleting TimingRecommender?

Reply:
It does not make sense for the LocationRecommender to obtain people's timetables. It should be the TimingRecommender's job to do it.

Reply:
@mynameizzhafeez Sorry, your comment is potentially very misleading. You should be committing often on your fork/PR so that I (as a collaborator) can see what you are referring to. Sounds to me like you want me to delete a huge chunk of code I spent too long doing. Please clarify.

Reply:
> It does not make sense for the LocationRecommender to obtain people's timetables. It should be the TimingRecommender's job to do it.

The Location Recommender will be initialised with a set of persons, so that’s where we will get the timetables.  From there, we can test each timing recommended by the timingrecommender.

Reply:
Sorry about the confusion.